lib/spack/spack/cmd/buildcache.py

```diff
@@ -163,15 +152,19 @@ def match_downloaded_specs(pkgs, allow_multiple_matches=False, force=False):
         matches = []
         tty.msg("buildcache spec(s) matching %s \n" % pkg)
         for spec in sorted(specs):
-            if spec.satisfies(pkg):
-                matches.append(spec)
+            if pkg.startswith('/'):
+                pkghash = pkg.replace('/', '')
+                if spec.dag_hash(7) == pkghash:
```

Comment:
IMO it would be worth doing a looser comparison here if the user provided a hash that was not exactly 7 characters in length. Something like

```
if pkghash and spec.dag_hash().startswith(pkghash):
```

Reply:
@gartung either this change should be made or the doc should be updated to mention that you should only supply exactly 7 characters of the hash

Reply:
OK. I was away from my computer for an appointment.